core/domain/user_services_test.py

```diff
@@ -4144,3 +4147,204 @@ def test_remove_question_submit_rights(self) -> None:
+        self.assertTrue(user_services.check_user_is_coordinator(
+            user_id_c, 'en'
+        ))
+
+        user_services.deassign_coordinator(
+            self.user_admin, user_c, 'en'
+        )
+
+        self.assertFalse(user_services.check_user_is_coordinator(
+            user_id_c, 'en'
+        ))
+
+    def test_non_admin_cannot_deassign_roles(self) -> None:
+        with self.assertRaisesRegex(
+            Exception,
+            'UnauthorizedUserException: Could not assign new role.'):
+            user_services.deassign_coordinator(
+                self.user_b, self.user_a, 'en')
+
+    def test_guest_user_cannot_deassign_roles(self) -> None:
+        guest_user = user_services.get_user_actions_info(None)
+        with self.assertRaisesRegex(
+            Exception,
+            'Guest user is not allowed to deassign roles to a user.'
+        ):
+            user_services.deassign_coordinator(
+                guest_user, self.user_b, 'en')
+
+    def test_guest_user_cannot_be_deassgined(
+        self
+    ) -> None:
+        with self.assertRaisesRegex(
+            Exception,
+            'No model exists for provided language.'
+        ):
+            user_services.deassign_coordinator(
+                self.user_admin, self.user_a, 'no_model')
+
+    def test_deassigning_for_non_existing_language_model(
+        self
+    ) -> None:
+        guest_user = user_services.get_user_actions_info(None)
+        with self.assertRaisesRegex(
+            Exception,
+            'Cannot change the role of the Guest user.'
+        ):
+            user_services.deassign_coordinator(
+                self.user_admin, guest_user, 'en')
+
+    def test_deassigning_role_from_non_coordinator(self) -> None:
+        with self.assertRaisesRegex(
+            Exception, 'This user is not a coordinator for this language'):
+            user_services.deassign_coordinator(
+                self.user_admin, self.user_a, 'en')
+
+    def test_get_translation_rights_from_model(self) -> None:
+        model = suggestion_models.TranslationCoordinatorsModel.get('en')
+        assert model is not None
+        model_object = user_services.get_translation_rights_from_model(
+            model)
+
+        # Asserting here because we have created a model for 'en' in setup.
+        assert model_object is not None
+        self.assertEqual(model.id, model_object.language_id)
+        self.assertEqual(
+            model.coordinators_count, model_object.coordinators_count)
+        self.assertEqual(
+            model.coordinator_ids, model_object.coordinator_ids)
+
+    def test_deassign_user_from_all_languages(self) -> None:
+        self.signup('c@example.com', 'C')
+        user_id_c = self.get_user_id_from_email('c@example.com')
+        self.set_translation_coordinators(['C'], 'en')
+        self.set_translation_coordinators(['C'], 'hi')
+        self.assertEqual(2, len(
+            user_services.get_translation_rights_with_user(user_id_c)))
```

Comment:
Remove this assert statement. If we want to ensure that calling `set_translation_coordinators` twice on the same user(s) assigns said user(s) with both translation coordinator languages, then write a separate test for `set_translation_coordinators`. For details about this test, please see my review comment in the `core/tests/test_utils.py` file.

Reply:
Sounds about right. Done.